lib/config/configuration.go

```diff
@@ -321,6 +329,111 @@ func ApplyFileConfig(fc *FileConfig, cfg *service.Config) error {
 			}
 		}
 	}
+	// read 'trusted_clusters' section:
+	if fc.Auth.Enabled() && len(fc.Auth.TrustedClusters) > 0 {
+		if err := readTrustedClusters(fc.Auth.TrustedClusters, cfg); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+	return nil
+}
+
+// readTrustedClusters parses the content of "trusted_clusters" YAML structure
+// and modifies Teleport 'conf' by adding "authorities" and "reverse tunnels"
+// to it
+func readTrustedClusters(clusters []TrustedCluster, conf *service.Config) error {
+	if len(clusters) == 0 {
+		return nil
+	}
+	// parseCAKey() gets called for every line in a "CA key file" which is
+	// the same as 'known_hosts' format for openssh
+	parseCAKey := func(bytes []byte) (*services.CertAuthority, error) {
```

Comment:
can you split this function into several? it's really hard to read as it spans several screens